variables.tf

```diff
@@ -56,6 +66,17 @@ An object describing the Storage Account to associate with the resource. This in
 DESCRIPTION
 }
 
+variable "associated_vnet" {
```

Comment:
YOu should be passing in subnets and not VNnets that way you know what you would need to associate and peer with your current resources and crete the private links.